kano_video/logic/youtube.py

```diff
@@ -45,34 +57,62 @@ def get_last_search_count():
 
 def search_youtube_by_keyword(keyword=None, popular=False, max_results=10,
                               start_index=1, parent_control=False):
-    url = 'http://gdata.youtube.com/feeds/api/videos'
-    params = {
-        'v': 2,
-        'vq': keyword,
-        'racy': 'exclude',
-        'orderby': 'relevance',
-        'alt': 'json',
-        'max-results': max_results,
-        'start-index': start_index
-    }
-    if popular:
-        params['orderby'] = 'viewCount'
 
-    if parent_control is True:
-        params['safeSearch'] = 'strict'
+#----Declare variables -------------
+    global prevPageIndex
 
-    success, error, data = requests_get_json(url, params=params)
+    str_safeSearch="none"
+    str_order = "relevance"
+    str_pageToken = ""
+#-----------------------------------
 
-    if not success:
-        logger.error('Searching YouTube by keyword failed: ' + error)
+
+    youtube = build(YOUTUBE_API_SERVICE_NAME, YOUTUBE_API_VERSION, #ADDED LINE INIATE OBJECT
+      developerKey=DEVELOPER_KEY)
+
+    if parent_control is True:
+        str_safeSearch = "strict"
+
+    #if popular: #If popular is set it seems like it only returns one video, we will leave that for another day...
+     #   str_order = "viewCount"
+
+#Check to see if we should navigate back or next and set str_pageToken
```

Comment:
@cwilson-github Can you ensure that all comments are at the same indentation level as the code that they are referring to so that it is easier to follow the program flow?